packages/compass-components/src/components/guide-cue/guide-cue.tsx

```diff
@@ -0,0 +1,200 @@
+import React, {
+  useEffect,
+  useCallback,
+  useRef,
+  useMemo,
+  useState,
+} from 'react';
+import { guideCueService, type ShowCueEventDetail } from './guide-cue-service';
+import { GuideCue as LGGuideCue } from '@leafygreen-ui/guide-cue';
+import type { GroupName } from './guide-cue-groups';
+
+const getDataCueId = ({
+  cueId,
+  groupId,
+}: {
+  cueId: string;
+  groupId?: string;
+}): string => {
+  return `guide-cue-${cueId}${groupId ? `-${groupId}` : ''}`;
+};
+
+type LGGuideCueProps = React.ComponentProps<typeof LGGuideCue>;
+
+type GroupAndStep =
+  | {
+      groupId: GroupName;
+      step: number;
+    }
+  | {
+      groupId?: never;
+      step?: never;
+    };
+
+// omit the props we are handling
+export type GuideCueProps<T> = Omit<
+  LGGuideCueProps,
+  | 'currentStep'
+  | 'refEl'
+  | 'numberOfSteps'
+  | 'onDismiss'
+  | 'open'
+  | 'onPrimaryButtonClick'
+  | 'setOpen'
+> &
+  GroupAndStep & {
+    cueId: string;
+    trigger: ({
+      ref,
+    }: {
+      ref: React.MutableRefObject<T | null>;
```

Comment:
This should rather be `LegacyRef<T>` or at least `Ref<T>` to match React html elements ref types better (otherwise you'll struggle to pass it directly to the element)